src/main/java/com/thebois/models/beings/roles/RoleFactory.java

```diff
@@ -0,0 +1,101 @@
+package com.thebois.models.beings.roles;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * Creates requested instances of specific roles.
+ */
+public final class RoleFactory {
+
+    private RoleFactory() {
+    }
+
+    /**
+     * Gets a role using its type.
+     *
+     * @param roleType The type of role to get
+     *
+     * @return The role with the specified type
+     *
+     * @throws java.util.NoSuchElementException If there is no such role
+     */
+    public static AbstractRole fromType(final RoleType roleType) {
+        return all()
+            .stream()
+            .filter(role -> role.getType().equals(roleType))
+            .findFirst()
+            .orElseThrow();
+    }
+
+    /**
+     * Creates an instance of every type of role.
+     *
+     * @return All roles
+     */
+    public static Collection<AbstractRole> all() {
+        final ArrayList<AbstractRole> roles = new ArrayList<>();
```

Comment:
```java
public static Collection<AbstractRole> all() {
  final Collection<AbstractRole> roles = new ArrayList<>();
  return roles;
```
is equivalent to
```java
public static Collection<AbstractRole> all() {
  final ArrayList<AbstractRole> roles = new ArrayList<>();
  return roles;
```
because in both cases the static type of the returned object is `Collection<AbstractRole>` and the dynamic type is `ArrayList<AbstractRole>`. 

It's just a matter of style. The benefit of keeping it as an `ArrayList` until it is returned is that the method has internal access to more methods to operate on the list. Which I think is good, right?